.github/workflows/pre-merge.yml

```
@@ -0,0 +1,142 @@
+---
+name: pre-merge
+
+on:
+  push:
+    branches:
+      - 'user/**'
+      - 'feature/**'
+      - 'improvement/**'
+      - 'dependabot/**'
+      - 'bugfix/**'
+      - 'w/**'
+      - 'q/**'
+
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - name: Install deps
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y tox \
+            libffi-dev \
+            libmysqlclient-dev \
+            libssl-dev
```

Use container with the existing docker image could be a good idea. With less install each time and only test command in the workflow will make the file more readable.

```suggestion
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
        with:
          python-version: '3.8'
      - name: Install deps
        run: |
          sudo apt-get update
          sudo apt-get install -y tox \
            libffi-dev \
            libmysqlclient-dev \
            libssl-dev
```

---

it will make the file more readable, that's for sure. But did we figured out a way of caching the docker image with actions? Because currently we only know how to rebuild it everytime (as far as I noticed at least). And if so, this will not bring any value to the build, and could actually make things worst. Because those packages are being downloaded from azure ubuntu repositories (Which are located close to the machines) where if we use docker and do not use any kind of caching, we could "lose" this nice environment that we have here.

Also note that we cannot use docker on every job, some require to be run inside a VM (docker and kube tests) where we must leave things this way.


Another option could be to create a small composite action for those and leave it inside the repo here and use it like the following:
```yaml
- name: Install deps
  uses: ./.github/actions/install-deps
```
Let me know what you think